### PR TITLE
Documentation explainability

### PR DESCRIPTION
**Pull request by PoorvaGarg**
~~Blocked by #557~~

This is a pull request to improve upon the documentation for usage of explainability module in models with binary random variables. It improves the documentation on three main points:
1. uses soft conditioning for conditioning on deterministic sites
2. changes computation of degree of responsibility
3. adds examples for additional queries

Some of the code, now uses masking on auxiliary sites to get the desired results which seemed necessary to be able to subselect the relevant samples.

### CODE REVIEW

File: docs/source/responsibility.ipynb
Comment (eb8680) [inline docs/source/responsibility.ipynb:1]: Should `responsibility.ipynb` be included in this PR? I thought it was just a scratch file for our discussion yesterday afternoon.

File: docs/source/explainable_categorical_alternate.ipynb
Comment (eb8680) [inline docs/source/explainable_categorical_alternate.ipynb:1]: Can you explain the difference between the existing notebook `explainable_categorical` and this new notebook `explainable_categorical_alternative`? It seems like the content mostly overlaps, is this file intended to replace `explainable_categorical` or complement it?

**Comment by PoorvaGarg** [comment]
This is a draft PR and wasn't ready for review. Thanks for the comments though. The final intended changes are only pertaining to the corrections needed in `explainable_categorical.ipynb`

**Review by eb8680 — APPROVED** [review]
Thanks for improving this!
